libs/notification/src/services/notification.service.ts

```diff
@@ -0,0 +1,48 @@
+import {Inject, Injectable} from '@angular/core';
+import {fromEvent, Observable, throwError} from 'rxjs';
+import {takeUntil} from 'rxjs/operators';
+
+import {NOTIFICATION_SUPPORT} from '../tokens/support';
+
+const NOT_SUPPORTED_ERROR$ = throwError(
+    () => new Error('Notification API is not supported in your browser'),
+);
+
+@Injectable({
+    providedIn: 'root',
+})
+export class NotificationService {
+    constructor(@Inject(NOTIFICATION_SUPPORT) private readonly support: boolean) {}
+
+    requestPermission(): Observable<NotificationPermission> {
+        if (!this.support) {
+            return NOT_SUPPORTED_ERROR$;
+        }
+
+        /**
+         * TODO: replace deprecated callback with promise after Safari 15+ support
+         * return from(Notification.requestPermission());
+         */
+        return new Observable(subscriber => {
+            void Notification.requestPermission(permission => {
+                subscriber.next(permission);
+                subscriber.complete();
+            })?.catch(err => subscriber.error(err));
+        });
+    }
+
+    open(title: string, options?: NotificationOptions): Observable<Notification> {
+        if (!this.support) {
+            return NOT_SUPPORTED_ERROR$;
+        }
+
+        const notification = new Notification(title, options);
```

Comment:
This would open notification immediately, right? I believe we should only create it inside `new Observable` so it is not shown unless you subscribe.

Reply:
My mistake, thanks)

✅